storm-core/src/jvm/org/apache/storm/command/AdminCommands.java

```diff
@@ -0,0 +1,133 @@
+
+        if (args.length == 0) {
+            throw new IllegalArgumentException("Missing command.");
+        }
+        initialize();
+        String command = args[0];
+        String[] newArgs = Arrays.copyOfRange(args, 1, args.length);
+        switch (command) {
+            case "remove_corrupt_topologies":
+                removeCorruptTopologies();
+                break;
+            default:
+                throw new RuntimeException("" + command + " is not a supported admin command");
+        }
+
+    }
+
+    private static void initialize() {
+        Map conf = ConfigUtils.readStormConfig();
+        BlobStore nimbusBlobStore = Utils.getNimbusBlobStore (conf, NimbusInfo.fromConf(conf));
+        List<String> servers = (List<String>) conf.get(Config.STORM_ZOOKEEPER_SERVERS);
+        Object port = conf.get(Config.STORM_ZOOKEEPER_PORT);
+        List<ACL> acls = null;
+        if (Utils.isZkAuthenticationConfiguredStormServer(conf)) {
+            acls = adminZkAcls();
+        }
+        try {
+            IStormClusterState stormClusterState = ClusterUtils.mkStormClusterState(conf, acls, new ClusterStateContext(DaemonType.NIMBUS));
+        } catch (Exception e) {
+            LOG.error("admin can't create stormClusterState");
+            new RuntimeException(e);
+        }
+        CuratorFramework zk = Zookeeper.mkClient(conf, servers, port, "", new DefaultWatcherCallBack(),conf);
+    }
+
+    // we might think of moving this method in Utils class
+    private static List<ACL> adminZkAcls() {
+        final List<ACL> acls = new ArrayList<>();
+        acls.add(ZooDefs.Ids.CREATOR_ALL_ACL.get(0));
+        acls.add(new ACL((ZooDefs.Perms.READ ^ ZooDefs.Perms.CREATE), ZooDefs.Ids.ANYONE_ID_UNSAFE));
+        return acls;
+    }
+
+    private static Set<String> getKeyListFromId( String corruptId) {
+        Set<String> keyLists = new HashSet<>();
+        keyLists.add(ConfigUtils.masterStormCodeKey(corruptId));
+        keyLists.add(ConfigUtils.masterStormConfKey(corruptId));
+        if(!ConfigUtils.isLocalMode(conf)) {
+            ConfigUtils.masterStormJarKey(corruptId);
+        }
+        return keyLists;
+    }
+
+    private static void removeCorruptTopologies( ) {
+        Iterator<String> corruptTopologies = listCorruptTopologies();
+        while(corruptTopologies.hasNext()) {
+            String corruptId = corruptTopologies.next();
+            stormClusterState.removeStorm(corruptId);
+            if(nimbusBlobStore instanceof LocalFsBlobStore) {
+                Iterator<String> blobKeys = getKeyListFromId(corruptId).iterator();
+                while(blobKeys.hasNext()) {
+                    stormClusterState.removeBlobstoreKey(blobKeys.next());
+                }
+            }
+        }
+    }
+
+    private static Iterator<String> listCorruptTopologies() {
+        Set<String> blobStoreTopologyIds = nimbusBlobStore.filterAndListKeys(new KeyFilter<String>() {
```

Comment:
Thanks for providing these steps. I fixed the issue and updated the pull request for review. Please take a look. 
However I had to change storm.cmd and storm-config.cmd as well to make it working. Please review these changes as well.